x-pack/plugins/security_solution/public/common/components/event_details/reason.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiTextColor, EuiFlexItem, EuiSpacer, EuiHorizontalRule, EuiTitle } from '@elastic/eui';
+import React, { useMemo } from 'react';
+
+import styled from 'styled-components';
+import { getRuleDetailsUrl, useFormatUrl } from '../link_to';
+import * as i18n from './translations';
+import { TimelineEventsDetailsItem } from '../../../../common';
+import { LinkAnchor } from '../links';
+import { useKibana } from '../../lib/kibana';
+import { APP_ID, SecurityPageName } from '../../../../common/constants';
+import { EVENT_DETAILS_PLACEHOLDER } from '../../../timelines/components/side_panel/event_details/translations';
+import { getFieldValue } from '../../../detections/components/host_isolation/helpers';
+
+interface Props {
+  data: TimelineEventsDetailsItem[];
+  eventId: string;
+}
+
+export const Indent = styled.div`
+  padding: 0 8px;
+  word-break: break-word;
+  line-height: 1.7em;
+`;
+
+export const ReasonComponent: React.FC<Props> = ({ eventId, data }) => {
+  const { navigateToApp } = useKibana().services.application;
+  const { formatUrl } = useFormatUrl(SecurityPageName.rules);
+
+  const reason = useMemo(
+    () => getFieldValue({ category: 'signal', field: 'signal.reason' }, data),
```

Comment:
We're going to want to make this `kibana.alert.reason` and use the constant here, but we can replace all of that as a wholesale change once the shared constants file is created